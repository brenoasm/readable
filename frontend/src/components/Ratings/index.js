import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import RateIcon from '../icons/RateIcon';

const propTypes = {
  voteScore: PropTypes.number.isRequired,
  rateUp: PropTypes.func.isRequired,
  rateDown: PropTypes.func.isRequired
};

const defaultProps = {
  voteScore: 0,
  rateUp: () => {},
  rateDown: () => {}
};

const Ratings = ({
  voteScore,
  rateUp,
  rateDown
}) => (
  <Fragment>
    <RateIcon
      onClick={rateUp}
      title="Gostei!"
    />
    <span>
      {voteScore}
    </span>
    <RateIcon
      onClick={rateDown}
      title="NÃ£o gostei!"
      rateDown
    />
  </Fragment>
);

Ratings.propTypes = propTypes;
Ratings.defaultProps = defaultProps;

export default Ratings;
