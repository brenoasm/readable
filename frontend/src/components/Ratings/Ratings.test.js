/* eslint-disable */

import React from 'react';

import Ratings from './';

describe('Ratings', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Ratings />);
    expect(wrapper).toMatchSnapshot();
  })
});
