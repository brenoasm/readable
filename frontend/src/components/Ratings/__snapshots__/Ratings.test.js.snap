// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ratings
    rateDown={[Function]}
    rateUp={[Function]}
    voteScore={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <RateIcon
          onClick={[Function]}
          rateDown={false}
          title="Gostei!"
/>,
        <span />,
        <RateIcon
          onClick={[Function]}
          rateDown={true}
          title="N達o gostei!"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "rateDown": false,
          "title": "Gostei!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
        },
        "ref": null,
        "rendered": 0,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "rateDown": true,
          "title": "N達o gostei!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <RateIcon
            onClick={[Function]}
            rateDown={false}
            title="Gostei!"
/>,
          <span />,
          <RateIcon
            onClick={[Function]}
            rateDown={true}
            title="N達o gostei!"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "rateDown": false,
            "title": "Gostei!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "rateDown": true,
            "title": "N達o gostei!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
